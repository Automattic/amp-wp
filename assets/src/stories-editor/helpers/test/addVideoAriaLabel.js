/**
 * Internal dependencies
 */
import { addVideoAriaLabel } from '../';

const getFigure = ( caption ) => (
	<figure>
		<amp-video>
			Fallback content
		</amp-video>
		{ caption && <figcaption>{ caption }</figcaption> }
	</figure>
);

describe( 'addVideoAriaLabel', () => {
	it( 'ignores arbitrary content', () => {
		const result = addVideoAriaLabel(
			<div>Ignore me</div>,
			{ name: 'anythin' },
			{},
		);

		expect( result ).toMatchSnapshot();
	} );

	it( 'ignores video content without an aria label attribute', () => {
		const result = addVideoAriaLabel(
			<div>Ignore me 2</div>,
			{ name: 'core/video' },
			{ ampAriaLabel: '' },
		);

		expect( result ).toMatchSnapshot();
	} );

	it( 'ignores video content even with an aria label attribute if content does not have the right format', () => {
		const result = addVideoAriaLabel(
			<div>Ignore me 3</div>,
			{ name: 'core/video' },
			{ ampAriaLabel: 'Ignored label' },
		);

		expect( result ).toMatchSnapshot();
	} );

	it( 'adds an aria label to video content without a caption', () => {
		const result = addVideoAriaLabel(
			getFigure(),
			{ name: 'core/video' },
			{ ampAriaLabel: 'Aria label 1' },
		);

		expect( result ).toMatchSnapshot();
	} );

	it( 'adds an aria label to video content with a caption', () => {
		const result = addVideoAriaLabel(
			getFigure( 'My video caption' ),
			{ name: 'core/video' },
			{ ampAriaLabel: 'Aria label 2' },
		);

		expect( result ).toMatchSnapshot();
	} );
} );
