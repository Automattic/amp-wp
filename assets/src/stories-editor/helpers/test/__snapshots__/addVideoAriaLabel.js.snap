// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addVideoAriaLabel adds an aria label to video content with a caption 1`] = `
<figure>
  <amp-video
    aria-label="Aria label 2"
  >
    Fallback content
  </amp-video>
  <figcaption>
    My video caption
  </figcaption>
</figure>
`;

exports[`addVideoAriaLabel adds an aria label to video content without a caption 1`] = `
<figure>
  <amp-video
    aria-label="Aria label 1"
  >
    Fallback content
  </amp-video>
</figure>
`;

exports[`addVideoAriaLabel ignores arbitrary content 1`] = `
<div>
  Ignore me
</div>
`;

exports[`addVideoAriaLabel ignores video content even with an aria label attribute if content does not have the right format 1`] = `
<div>
  Ignore me 3
</div>
`;

exports[`addVideoAriaLabel ignores video content without an aria label attribute 1`] = `
<div>
  Ignore me 2
</div>
`;
